src/db/mysql/models.rs

```diff
@@ -49,19 +50,27 @@ pub fn run_embedded_migrations(settings: &Settings) -> Result<()> {
     Ok(embedded_migrations::run(&conn)?)
 }
 
+enum CollectionLock {
+    Read,
+    Write,
+}
+
 /// Per session Db metadata
 pub(super) struct MysqlDbSession {
     /// The "current time" on the server used for this session's operations
     pub(super) timestamp: i64,
```

Comment:
Is this in seconds, milliseconds, etc? The timestamp we get from clients is f64, with two digits of precision.

Reply:
modified is stored in the db as milliseconds https://github.com/mozilla-services/syncstorage-rs/blob/master/migrations/2018-08-28-010336_init/up.sql#L15

the storage layer doesn't have any direct interaction with client supplied timestamps. the modified field on BSOs is always set by the server. the handler layer gets timestamps back from the database and handles the comparisons against the client supplied headers.

Though the one thing the python db code does is make its result timestamps floats (via bigint2ts). I'll worry about that when we start integrating the db into the handlers (I think it might make more sense for the handler to make the conversion/rendering into a float as both mysql/spanner are likely storing int64s)

Reply:
Strong 👍 to treating the "float with two-decimal places" encoding as a kind of legacy encoding issue and dealing with it as close to the edge as possible.

Reply:
Should we immediately convert to milliseconds and use that everywhere internally then? Will existing clients handle 3 digits for milliseconds in the Last-(Un)Modified?

Reply:
> Will existing clients handle 3 digits for milliseconds in the Last-(Un)Modified?

Probably, but I wouldn't count on it :-/

I was actually remembering incorrectly here, the python server uses a quantized `Decimal` instance (not floats) as its canonical timestamp format internally:

  https://github.com/mozilla-services/server-syncstorage/blob/master/syncstorage/util.py#L13

Integer milliseconds with the lowest digit always zero will probably be an easier format to deal with in rust code.